src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs

```diff
@@ -226,6 +226,10 @@ private ImmutableArray<ISymbol> GetSymbolsForExpressionOrStatementContext()
                 ? _context.SemanticModel.LookupStaticMembers(_context.LeftToken.SpanStart)
                 : _context.SemanticModel.LookupSymbols(_context.LeftToken.SpanStart);
 
+            // filter our top level locals if we're inside a type declaration.
+            if (_context.ContainingTypeDeclaration != null)
+                symbols = symbols.WhereAsArray(s => s.ContainingSymbol.Name != WellKnownMemberNames.TopLevelStatementsEntryPointMethodName);
```

Comment:
Why `LookupSymbols` is returning the extra symbols in the first place?

Reply:
> Why LookupSymbols is returning the extra symbols in the first place?

I forget the exact reason, but it's int eh LDM notes.  I think it's because if we *want* to add the facility for some top level symbols we can do so in a way that is sensible (e.g. imagine top level local functions in namespaces).  @333fred may remember better.  There's a subtle difference between not-visible, and visible-but-not-usable, and i recall there being reasons we wanted the latter.